File: backend/internal/app/router/ai_judge_config.go
package router

import (
	"backend/internal/app/api"
	"backend/internal/app/middleware"
	"github.com/gin-gonic/gin"
)

func InitAiJudgeConfigRouter(Router *gin.RouterGroup) {
	aiJudgeConfigRouter := Router.Group("ai-judge-config").Use(middleware.JWTAuth())
	{
		aiJudgeConfigRouter.GET("list", api.GetAiJudgeConfigList)                  // 获取配置列表
		aiJudgeConfigRouter.POST("create", api.CreateAiJudgeConfig)                // 创建配置
		aiJudgeConfigRouter.POST("update", api.UpdateAiJudgeConfig)                // 更新配置
		aiJudgeConfigRouter.POST("delete", api.DeleteAiJudgeConfig)                // 删除配置
		aiJudgeConfigRouter.POST("toggle", api.ToggleAiJudgeConfig)                // 切换启用状态
		aiJudgeConfigRouter.GET("enabled", api.GetEnabledAiJudgeConfig)            // 获取当前启用的配置
		aiJudgeConfigRouter.POST("grade", api.AiGrade)                             // AI判题
		aiJudgeConfigRouter.POST("toggle-auto-grading", api.ToggleAutoGrading)    // 切换自动判题状态
		aiJudgeConfigRouter.GET("pending-list", api.GetPendingGradeList)           // 获取待判题列表
		aiJudgeConfigRouter.POST("batch-grade", api.BatchGrade)                    // 批量AI判题（自动提交）
		aiJudgeConfigRouter.GET("history", api.GetAiGradeHistory)                  // 获取AI判题历史
		aiJudgeConfigRouter.POST("batch-grade-preview", api.BatchGradePreview)    // 批量AI判题预览（不提交）
		aiJudgeConfigRouter.POST("submit-batch-grade", api.SubmitBatchGrade)      // 提交批量判题结果
	}
}

File: frontend/src/request/api/aiJudgeConfig.js
import service from '../index'

// 获取AI判题配置列表
export function getAiJudgeConfigList() {
    return service({
        url: '/ai-judge-config/list',
        method: 'get',
    })
}

// 创建AI判题配置
export function createAiJudgeConfig(data) {
    return service({
        url: '/ai-judge-config/create',
        method: 'post',
        data
    })
}

// 更新AI判题配置
export function updateAiJudgeConfig(data) {
    return service({
        url: '/ai-judge-config/update',
        method: 'post',
        data
    })
}

// 删除AI判题配置
export function deleteAiJudgeConfig(data) {
    return service({
        url: '/ai-judge-config/delete',
        method: 'post',
        data
    })
}

// 切换AI判题配置启用状态
export function toggleAiJudgeConfig(data) {
    return service({
        url: '/ai-judge-config/toggle',
        method: 'post',
        data
    })
}

// 获取当前启用的AI判题配置
export function getEnabledAiJudgeConfig() {
    return service({
        url: '/ai-judge-config/enabled',
        method: 'get',
    })
}

// AI判题
export function aiGrade(data) {
    return service({
        url: '/ai-judge-config/grade',
        method: 'post',
        data
    })
}

// 切换自动判题状态
export function toggleAutoGrading(data) {
    return service({
        url: '/ai-judge-config/toggle-auto-grading',
        method: 'post',
        data
    })
}

// 获取待判题列表
export function getPendingGradeList(params) {
    return service({
        url: '/ai-judge-config/pending-list',
        method: 'get',
        params
    })
}

// 批量AI判题
export function batchGrade(data) {
    return service({
        url: '/ai-judge-config/batch-grade',
        method: 'post',
        data
    })
}

// 获取AI判题历史
export function getAiGradeHistory(params) {
    return service({
        url: '/ai-judge-config/history',
        method: 'get',
        params
    })
}

// 批量AI判题预览（不直接提交）
export function batchGradePreview(data) {
    return service({
        url: '/ai-judge-config/batch-grade-preview',
        method: 'post',
        data
    })
}

// 提交批量判题结果
export function submitBatchGrade(data) {
    return service({
        url: '/ai-judge-config/submit-batch-grade',
        method: 'post',
        data
    })
}
